src/webpage/form_handling/ground_truth_handler.py

```
@@ -0,0 +1,124 @@
+        # Get current row and dataframe
+        df = st.session_state.get(get_results_key())
+        if df is None:
+            return {}
+
+        ind = df.index[st.session_state[SELECTED_ROW_ID]]
+
+        # Set default values from dataframe
+        relevance = df.loc[ind].get(SYN_QA_RELEVANCE, True)
+        irrelevant = st.checkbox(
+            "The question is not relevant",
+            key=f"syn_gt_qa_irrelevant_{self.question_hash}",
+            value=False if relevance else True,
+        )
+
+        question = df.loc[ind].get(SYN_CORRECTED_QUESTION)
+        if pd.isna(question):
+            question = df.loc[ind].get(QUESTION_COLUMN, "")
+
+        corrected_question = st.text_area(
+            "Rewrite the question (Optional)",
+            key=f"syn_corrected_question_{self.question_hash}",
+            value=question,
+        )
+
+        gt_code = df.loc[ind].get(SYN_GT_ANSWER)
+        if pd.isna(gt_code):
+            gt_code = df.loc[ind].get(PREDICTIONS_COLUMN, "")
+
+        gt_answer = st.text_area(
+            "Provide Ground Truth (Optional)",
+            key=f"syn_gt_code_{self.question_hash}",
+            value=gt_code,
+        )
+
+        # Return the collected data
+        return {
+            SYN_QA_RELEVANCE: not irrelevant,
+            SYN_CORRECTED_QUESTION: corrected_question,
+            SYN_GT_ANSWER: gt_answer,
+        }
+
+    def save_to_dataframe(self, df: Optional[pd.DataFrame] = None) -> pd.DataFrame:
+        """
+        Save the ground truth data to the dataframe.
+
+        Args:
+            df (Optional[pd.DataFrame]): The dataframe to modify. If None, retrieves from session state.
+
+        Returns:
+            pd.DataFrame: The updated dataframe
+        """
+        if df is None:
+            df = st.session_state.get(get_results_key())
+
+        if df is None:
+            st.error("Results dataframe not found in session state")
+            return pd.DataFrame()
```

```
df = df or st.session_state.get(get_results_key())
if df is None:
    st.error("Results dataframe not found in session state")
    return pd.DataFrame()
```

---

I think this should raise an error if dataframe is not empty:`ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().`